pkg/loadbalance/consist.go

```diff
@@ -483,9 +483,9 @@ func (cb *consistBalancer) Rebalance(change discovery.Change) {
 	}
 	// TODO: Use TreeMap to optimize performance when updating.
 	// Now, due to the lack of a good red-black tree implementation, we can only build the full amount once per update.
-	cb.updateLock.RLock()
+	cb.updateLock.Lock()
 	cb.updateConsistInfo(change.Result)
-	cb.updateLock.RUnlock()
+	cb.updateLock.Unlock()
```

Comment:
seems updateLock doesn't use RLock() anymore, maybe we should use sync.Mutex instead of sync.RWMutex